### PR TITLE
Load more chat threads implementation.

### PR DESCRIPTION
**Pull request by seekdavidlee**
This PR will load the initial chat threads based on `chatThreadsPerPage` property configured in the ChatList component. The response will return a nextLink that will tell us if there are going to be more pages to load. If so, we enable "load more" button. The request is already configured to return chat threads in the right order. 

When the user clicks on "load more", we can take the existing count and add it to their desired number of chat threads to load and use this to calculate the new number of chat threads to load. Note that we cannot exceed more than 50 chat threads per page - per the documentation, so we have a way to control that to page through to get to our desired count.

### CODE REVIEW

**Comment by plasne** [comment]
At least for a time, pages don't necessarily return the full page count because there are "soft-deleted" threads. You can test this by deleting a chat thread and then reloading. We should probably use the same code you used in load more (ie. paging) for the initial load in order to ensure we actually get the right number.

**Comment by seekdavidlee** [comment]
> At least for a time, pages don't necessarily return the full page count because there are "soft-deleted" threads. You can test this by deleting a chat thread and then reloading. We should probably use the same code you used in load more (ie. paging) for the initial load in order to ensure we actually get the right number.

I deleted a chat thread and then clicked on load more. And it did not load the chat thread that I as the user deleted. The same code for paging through is used in both the initial load and load more.

**Comment by TechnicallyWilliams** [comment]
> > At least for a time, pages don't necessarily return the full page count because there are "soft-deleted" threads. You can test this by deleting a chat thread and then reloading. We should probably use the same code you used in load more (ie. paging) for the initial load in order to ensure we actually get the right number.
> 
> I deleted a chat thread and then clicked on load more. And it did not load the chat thread that I as the user deleted. The same code for paging through is used in both the initial load and load more.

Not sure what the steps to repeat are but I deleted a chat thread in teams web then hit load more in my local app and the deleted chat still showed up.

**Comment by plasne** [comment]
> > At least for a time, pages don't necessarily return the full page count because there are "soft-deleted" threads. You can test this by deleting a chat thread and then reloading. We should probably use the same code you used in load more (ie. paging) for the initial load in order to ensure we actually get the right number.
> 
> I deleted a chat thread and then clicked on load more. And it did not load the chat thread that I as the user deleted. The same code for paging through is used in both the initial load and load more.

I was probably not clear. You don't get the DELETED item in the return, but you also don't get 10 items (or whatever you requested), you get that minus the deletion count.

Having said that, you are calling the same code path (I missed that before) so it should page and pull more anyway. All good.

File: packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts [hunk 1]
```diff
@@ -325,13 +350,7 @@ class StatefulGraphChatListClient implements StatefulClient<GraphChatListClient>
...
-          if (this._graph !== undefined) {
-            loadChatThreads(this._graph, this.chatThreadsPerPage).then(
-              chats => this.handleChatThreads(chats),
-              err => error(err)
-            );
-          }
+          this.loadAndAppendChatThreads('', [], this.chatThreadsPerPage);
         }
         return;
       case ProviderState.SignedOut:
```
Comment (plasne) [inline packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts:356]: If we do a sign-out, shouldn't we clear the chat list also?

Comment (seekdavidlee) [reply packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts:356]: if we do a sign out, we should get redirected out, the entire component should be refreshed.

**Comment by seekdavidlee** [comment]
> > > At least for a time, pages don't necessarily return the full page count because there are "soft-deleted" threads. You can test this by deleting a chat thread and then reloading. We should probably use the same code you used in load more (ie. paging) for the initial load in order to ensure we actually get the right number.
> > 
> > 
> > I deleted a chat thread and then clicked on load more. And it did not load the chat thread that I as the user deleted. The same code for paging through is used in both the initial load and load more.
> 
> I was probably not clear. You don't get the DELETED item in the return, but you also don't get 10 items (or whatever you requested), you get that minus the deletion count.
> 
> Having said that, you are calling the same code path (I missed that before) so it should page and pull more anyway. All good.

When load more is clicked on, it is always started from a new paging process. If so, I am still not clear on why it would miss any chat threads.

**Comment by plasne** [comment]
> > > > At least for a time, pages don't necessarily return the full page count because there are "soft-deleted" threads. You can test this by deleting a chat thread and then reloading. We should probably use the same code you used in load more (ie. paging) for the initial load in order to ensure we actually get the right number.
> > > 
> > > 
> > > I deleted a chat thread and then clicked on load more. And it did not load the chat thread that I as the user deleted. The same code for paging through is used in both the initial load and load more.
> > 
> > 
> > I was probably not clear. You don't get the DELETED item in the return, but you also don't get 10 items (or whatever you requested), you get that minus the deletion count.
> > Having said that, you are calling the same code path (I missed that before) so it should page and pull more anyway. All good.
> 
> When load more is clicked on, it is always started from a new paging process. If so, I am still not clear on why it would miss any chat threads.

With the paging it's fine. I am good with it.

**Review by plasne — APPROVED** [review]

**Review by andrewDoing — APPROVED** [review]

**Pull request merged**